aspnetcore/tutorials/razor-pages/razor-pages-start/sample/RazorPagesMovie60/ProgramProd.cs

```diff
@@ -0,0 +1,37 @@
+//#define PROD
+#if PROD
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.DependencyInjection;
+
+var builder = WebApplication.CreateBuilder(args);
+
+builder.Services.AddRazorPages();
+
+var app = builder.Build();
+
+if (app.Environment.IsDevelopment())
+{
+    builder.Services.AddDbContext<RazorPagesMovieContext>(options =>
+    options.UseSqlite(builder.Configuration.GetConnectionString("RazorPagesMovieContext")));
+    app.UseDeveloperExceptionPage();
+}
+else
+{
+    builder.Services.AddDbContext<RazorPagesMovieContext>(options 
+        => options.UseSqlServer(builder.Configuration.GetConnectionString("ProductionMovieContext")));
+
+    app.UseExceptionHandler("/Error");
+    app.UseHsts();
+}
```

Comment:
> @Rick-Anderson Would something like this work?

```
var builder = WebApplication.CreateBuilder(args);
Action<DbContextOptionsBuilder> dbAction;
builder.Services.AddDbContext<RazorPagesMovieContext>(dbAction);
```
Error	CS0165 Use of unassigned local variable `dbAction`

Reply:
You can check the environment using `builder.Environment`, so this could be split up into something like the following which adds the services before `Build()` and the middleware after:

```C#
var builder = WebApplication.CreateBuilder(args);

builder.Services.AddRazorPages();

if (builder.Environment.IsDevelopment())
{
    builder.Services.AddDbContext<RazorPagesMovieContext>(options =>
        options.UseSqlite(builder.Configuration.GetConnectionString("RazorPagesMovieContext")));
}
else
{
    builder.Services.AddDbContext<RazorPagesMovieContext>(options =>
        options.UseSqlServer(builder.Configuration.GetConnectionString("ProductionMovieContext")));
}

var app = builder.Build();

if (!app.Environment.IsDevelopment())
{
    app.UseExceptionHandler("/Error");
    app.UseHsts();
}
```

The developer exception page is added automatically in development by `WebApplication`, so I intentionally removed the `app.UseDeveloperExceptionPage()` call.

Reply:
Muchos gracious @halter73